### PR TITLE
51307 exporter configuration kebab case

### PR DESCRIPTION
**Pull request by ajanoni-camunda**
## Description

 This PR hardens exporter configuration binding and expands test coverage for configuration parsing in broker, Elasticsearch exporter, and Opensearch exporter modules.

  Motivation

  Exporter args are provided as raw maps and can come in multiple key/value shapes.
  Previously, binding was not reliably kebab-case compatible for these raw exporter args, so config keys could behave inconsistently depending on how they were written. This PR makes parsing robust for both kebab-case and camelCase keys, and improves fail-fast behaviour for invalid properties.

  Changes

  Feature changes

   - Added a new @StrictConfiguration annotation for exporter configuration classes.
   - Updated broker-side exporter configuration mapping to:
   - normalize camelCase keys to kebab-case before binding,
   - handle both kebab-case and camelCase inputs consistently,
   - support strict unknown-property rejection for classes annotated with @StrictConfiguration,
   - improve unknown-property error reporting.
   - Enhanced list deserialization behavior to support:
   - empty string values as empty lists,
   - single non-empty string values as singleton lists.
   - Applied @StrictConfiguration to:
   - ElasticsearchExporterConfiguration
   - OpensearchExporterConfiguration

  Test changes

   - Added/updated broker tests to cover:
   - key normalization behavior (kebab-case and camelCase),
   - strict unknown-property rejection,
   - indexed-list and list edge cases.
   - Expanded Elasticsearch exporter index configuration tests to validate mixed key styles and list encodings via ExporterConfiguration.fromArgs.
   - Expanded Opensearch exporter index configuration tests with the same mapping scenarios for parity.

  Result

   - Exporter configuration parsing is now consistent for both kebab-case and camelCase keys.
   - Invalid exporter properties fail fast for strict configs.
   - Test coverage now explicitly guards key parsing and list-shape compatibility across broker, Elasticsearch, and Opensearch exporter paths.

## Related issues

closes # https://github.com/camunda/camunda/issues/51307

### CODE REVIEW

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/context/ExporterConfiguration.java [hunk 1]
```diff
@@ -69,14 +92,71 @@ public <T> T instantiate(final Class<T> configClass) {
...
+          entry.getValue() instanceof final Map<?, ?> nested
+              ? normalizeKeys((Map<String, Object>) nested)
+              : entry.getValue();
+      result.put(key, value);
+    }
+    return result;
+  }
+
+  /** Converts a camelCase or PascalCase string to kebab-case. Kebab-case input is unchanged. */
+  private static String camelToKebab(final String name) {
```
[the remark this answers is not in the document]
Comment (ajanoni-camunda) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/context/ExporterConfiguration.java:175]: @copilot apply changes based on [this feedback](https://github.com/camunda/camunda/pull/51308#discussion_r3103273074)
